api/renewer.go

```diff
@@ -0,0 +1,280 @@
+package api
+
+import (
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+func init() {
+	// Seed the random generator
+	rand.Seed(time.Now().Unix())
```

Comment:
Given the high security context of Vault I'd be interested in hearing from @jefferai and team on this one, since it will affect all importers and might cause unintended side-effects. For most practical cases though I agree it's probably fine to just re-seed like you are doing already. We should probably switch it to a `UnixNano()` to slim the chances of a common seed down.